src/rum/instrumentation/DdRumErrorTracking.tsx

```diff
@@ -20,39 +21,87 @@ export class DdRumErrorTracking {
     // eslint-disable-next-line 
     private static defaultErrorHandler = (_error: any, _isFatal?: boolean) => {}
 
+    // eslint-disable-next-line 
+    private static defaultConsoleError = (..._params: unknown[]) => {}
+
     /**
      * Starts tracking errors and sends a RUM Error event every time an error is detected.
      */
     static startTracking(): void {
         // extra safety to avoid wrapping the Error handler twice
-        if (this.isTracking) {
+        if (DdRumErrorTracking.isTracking) {
+            console.log("DdRumErrorTracking already started");
             return
         }
         
         if (ErrorUtils) {
-            this.defaultErrorHandler = ErrorUtils.getGlobalHandler();
+            DdRumErrorTracking.defaultErrorHandler = ErrorUtils.getGlobalHandler();
+            DdRumErrorTracking.defaultConsoleError = console.error;
 
-            ErrorUtils.setGlobalHandler(this.onError);
+            
+            ErrorUtils.setGlobalHandler(DdRumErrorTracking.onGlobalError);
+            console.error = DdRumErrorTracking.onConsoleError;
 
-            this.isTracking = true;
+            DdRumErrorTracking.isTracking = true;
         }
 
     }
 
     // eslint-disable-next-line 
-    static onError(error: any, isFatal?: boolean) {
-        // Stack trace
-        let stack = EMPTY_STACK_TRACE
-        if (error.hasOwnProperty('componentStack')) {
-            stack = String(error.componentStack);
-        } else if (error.hasOwnProperty('sourceURL') && error.hasOwnProperty('line') && error.hasOwnProperty('column')) {
-            stack = "at " + error.sourceURL + ":" + error.line + ":" + error.column
+    static onGlobalError(error: any, isFatal?: boolean) {
+        DdRum.addError(
+            String(error), 
+            TYPE_SOURCE, 
+            DdRumErrorTracking.getErrorStackTrace(error), 
+            new Date().getTime(), 
+            { "_dd.error.is_crash": isFatal, "_dd.error.raw": error }
+        ).then(() => {
+            DdRumErrorTracking.defaultErrorHandler(error, isFatal);
+        });
+    }
+
+    static onConsoleError(...params: unknown[]) {
+        let stack: string = EMPTY_STACK_TRACE
+        for (let i = 0; i < params.length; i += 1) {
+            const param = params[i];
+            const paramStack = DdRumErrorTracking.getErrorStackTrace(param)
+            if (paramStack != undefined && paramStack != EMPTY_STACK_TRACE){
```

Comment:
I don't think you need the second condition as the default value is `EMPTY_STACK_TRACE`

Reply:
it's be be safer here. I don't trust JS